### PR TITLE
VB-1926 Update orchestration service to acquire full username and ret…

### PR DESCRIPTION
**Pull request by AledWEvans**
…urn it to the UI for each entry - added new end point to return visit history

## What does this pull request do?

allows us to return the history for a booking

## What is the intent behind these changes?

the intent is to allow users to see how did what and when

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt [hunk 1]
```diff
@@ -35,6 +35,14 @@ interface VisitRepository : JpaRepository<Visit, Long>, JpaSpecificationExecutor
   )
   fun findByReference(reference: String): Visit?
 
+  @Query(
+    "SELECT * FROM visit " +
+      "WHERE reference = :reference AND visit_status IN ('BOOKED','CANCELLED')  " +
+      "ORDER BY modify_timestamp",
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt:41]: should this be create_timestamp or will it not matter?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt [hunk 2]
```diff
@@ -35,6 +35,14 @@ interface VisitRepository : JpaRepository<Visit, Long>, JpaSpecificationExecutor
...
+    nativeQuery = true,
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt:42]: does this need to be a native query?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt [hunk 3]
```diff
@@ -35,6 +35,14 @@ interface VisitRepository : JpaRepository<Visit, Long>, JpaSpecificationExecutor
   )
   fun findByReference(reference: String): Visit?
 
+  @Query(
+    "SELECT * FROM visit " +
```
Comment (dhirajshettymoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt:39]: I think this query will also return the SUPERSEDED_CANCELLATION records. The query might need to be updated to excluded them.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/visit/VisitHistoryByReferenceTest.kt [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    visitEntityHelper.create(reference = reference, prisonerId = "FF0000AA", visitStatus = CHANGING, visitStart = visitTime)
+
+    // When
+    val responseSpec = callVisitHistoryByReference(webTestClient, reference, roleVisitSchedulerHttpHeaders)
+
+    // Then
+    responseSpec.expectStatus().isOk
+      .expectBody()
+      .jsonPath("$.length()").isEqualTo(2)
+      .jsonPath("$[0].reference").isEqualTo(reference)
```
Comment (dhirajshettymoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/visit/VisitHistoryByReferenceTest.kt:45]: should we also test for the status to ensure only BOOKED and CANCELLED are returned?  That will also help ensure that the records are returned in the right order.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt (hunk 3 above)
Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt:39]: SUPERSEDED_CANCELLATION is the outcome status

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt (hunk 2 above)
Comment (AledWEvans) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt:42]: it's easy this way

File: src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/visit/VisitHistoryByReferenceTest.kt (hunk 4 above)
Comment (rj-adams) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/visit/VisitHistoryByReferenceTest.kt:45]: Also, one should never assume database records will be returned in a specific order unless that order is specified. Suggest putting an"order by" clause in the SQL or whatever JPA requires to make that happen

Comment (AledWEvans) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/integration/visit/VisitHistoryByReferenceTest.kt:45]: There is an order on the SQL statment!

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt (hunk 3 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/VisitRepository.kt:39]: As discussed, yes we need to show these visits.

**Review by dhirajshettymoj — APPROVED** [review]
LGTM

**Pull request merged**